turbineControl_v2/src/program/turbineControl.lua

```diff
@@ -448,6 +448,16 @@ function findOptimalFuelRodLevel()
 
         --Find precise level
         while true do
+            if (controlRodLevel <= 0) then
+                -- prevent controlRodLevel from going negative in case the reactor cannot produce enough steam to reach targetSteamOutput
+                controlRodLevel = 0
+                r.setAllControlRodLevels(controlRodLevel)
+                rodLevel = controlRodLevel
+                saveOptionFile()
+                print("Target RodLevel: " .. controlRodLevel)
+                sleep(2)
+                break; 
```

Comment:
This code previously did not check whether the `controlRodLevel` had fallen to zero or below, and since its condition for quitting the loop is attaining `targetSteamOutput`, it would never quit because the reactor is unable to reach `targetSteamOutput` and will not do so no matter how many times `controlRodLevel` is reduced below zero.